### PR TITLE
add space createPg

### PR DESCRIPTION
**Pull request by yangxinyue-git**

### CODE REVIEW

**Comment by drizzlezyk** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by drizzlezyk** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

File: space/infrastructure/repositoryimpl/space.go [hunk 1]
```diff
@@ -0,0 +1,854 @@
...
+func (adapter *projectAdapter) listGlobalAndSortByDownloadCount(
+	do *repositories.GlobalResourceListDO) ([]ProjectSummaryDO, int, error) {
+	var items []projectDO
+	var count int64
+
+	// 基础查询条件
+	baseQuery := adapter.db()
+
+	// 计算总数
+	if err := baseQuery.Model(&projectDO{}).Count(&count).Error; err != nil {
```
[the remark this answers is not in the document]
Comment (yangxinyue-git) [reply space/infrastructure/repositoryimpl/space.go]: 已改正

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#10060;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>0.0
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/5c049736c2cd46ba9a4c8bbcbc5e0fca/12ce120ba2b34a8db2b7e125300e45e5">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>0.0
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/5c049736c2cd46ba9a4c8bbcbc5e0fca/0a6fdb939e26433688573d716d65f42e">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>0.0
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/5c049736c2cd46ba9a4c8bbcbc5e0fca/b1c7747989774d55864bc80fda5a8ac5">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>0.0
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/5c049736c2cd46ba9a4c8bbcbc5e0fca/5e5d095643604bda83c7c3c89d90a2aa">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>0.0
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/5c049736c2cd46ba9a4c8bbcbc5e0fca/c2f683eb9e914fadaec8f51be793d243">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>0.0
%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/5c049736c2cd46ba9a4c8bbcbc5e0fca/bf42abcda0fd477b9aa3e49abe4fd55a">点击跳转</a></td>
</tr>
</table>

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>测试覆盖率</td>
    <td>&#10060;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/xihe-server/756/756_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/5c049736c2cd46ba9a4c8bbcbc5e0fca/85eadbf2600b42b0b85f91e13363c360">点击跳转</a></td>
</tr>
</table>

**Pull request closed without merging**